Gems/Blast/Code/Tests/Editor/EditorBlastChunksAssetHandlerTest.cpp

```diff
@@ -0,0 +1,209 @@
+        MOCK_METHOD0(DisableCatalog, void());
+        MOCK_METHOD1(EnableCatalogForAsset, void(const AZ::Data::AssetType&));
+        MOCK_METHOD3(EnumerateAssets, void(BeginAssetEnumerationCB, AssetEnumerationCB, EndAssetEnumerationCB));
+        MOCK_METHOD1(GenerateAssetIdTEMP, AZ::Data::AssetId(const char*));
+        MOCK_METHOD1(GetAllProductDependencies, AZ::Outcome<AZStd::vector<AZ::Data::ProductDependency>, AZStd::string>(const AZ::Data::AssetId&));
+        MOCK_METHOD3(GetAllProductDependenciesFilter, AZ::Outcome<AZStd::vector<AZ::Data::ProductDependency>, AZStd::string>(const AZ::Data::AssetId&, const AZStd::unordered_set<AZ::Data::AssetId>&, const AZStd::vector<AZStd::string>&));
+        MOCK_METHOD1(GetAssetPathById, AZStd::string(const AZ::Data::AssetId&));
+        MOCK_METHOD1(GetDirectProductDependencies, AZ::Outcome<AZStd::vector<AZ::Data::ProductDependency>, AZStd::string>(const AZ::Data::AssetId&));
+        MOCK_METHOD1(GetHandledAssetTypes, void(AZStd::vector<AZ::Data::AssetType>&));
+        MOCK_METHOD0(GetRegisteredAssetPaths, AZStd::vector<AZStd::string>());
+        MOCK_METHOD2(InsertDeltaCatalog, bool(AZStd::shared_ptr<AzFramework::AssetRegistry>, size_t));
+        MOCK_METHOD2(InsertDeltaCatalogBefore, bool(AZStd::shared_ptr<AzFramework::AssetRegistry>, AZStd::shared_ptr<AzFramework::AssetRegistry>));
+        MOCK_METHOD1(LoadCatalog, bool(const char*));
+        MOCK_METHOD2(RegisterAsset, void(const AZ::Data::AssetId&, AZ::Data::AssetInfo&));
+        MOCK_METHOD1(RemoveDeltaCatalog, bool(AZStd::shared_ptr<AzFramework::AssetRegistry>));
+        MOCK_METHOD1(SaveCatalog, bool(const char*));
+        MOCK_METHOD0(StartMonitoringAssets, void());
+        MOCK_METHOD0(StopMonitoringAssets, void());
+        MOCK_METHOD1(UnregisterAsset, void(const AZ::Data::AssetId&));
+    };
+
+    class MockAssetManager
+        : public AZ::Data::AssetManager
+    {
+    public:
+        MockAssetManager(const AZ::Data::AssetManager::Descriptor& desc) :
+            AssetManager(desc)
+        {
+        }
+    };
+
+    class EditorBlastSliceAssetHandlerTestFixture
+        : public AllocatorsTestFixture
+    {
+    public:
+        AZStd::unique_ptr<UnitTest::MockComponentApplication> m_mockComponentApplicationBusHandler;
+        AZStd::unique_ptr<MockAssetCatalogRequestBusHandler> m_mockAssetCatalogRequestBusHandler;
+        AZStd::unique_ptr<MockAssetManager> m_mockAssetManager;
+        AZStd::unique_ptr<AZ::SerializeContext> m_serializeContext;
+
+        void SetUpSliceComponents()
+        {
+            m_serializeContext = AZStd::make_unique<AZ::SerializeContext>();
+
+            AZ::Entity::Reflect(m_serializeContext.get());
+            AzToolsFramework::Components::EditorComponentBase::Reflect(m_serializeContext.get());
+        }
+
+        void TearDownSliceComponents()
+        {
+            m_serializeContext.reset();
+        }
```

Comment:
this whole test file references "Slices" in several places  - should these be "Chunks"?

Reply:
You are right, Chunks are the new asset types.